### PR TITLE
Updated what is my ip

### PR DESCRIPTION
**Pull request by bfuzzy1**

### CODE REVIEW

File: tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go [hunk 1]
```diff
@@ -6,20 +6,77 @@ CREATED: 2023-01-21
...
+	err = json.NewDecoder(response.Body).Decode(&data)
+	if err != nil {
+		return "", 0, 0, err
+	}
+
+	return data.Query, data.Lat, data.Lon, nil
+}
+
+func searchShodan(ip string) {
+	url := "https://www.shodan.io/search?query=" + ip
```
Comment (privateducky) [inline tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go:40]: won't we get rate limited by Shodan?

File: tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go [hunk 2]
```diff
@@ -6,20 +6,77 @@ CREATED: 2023-01-21
...
+
+	Endpoint "github.com/preludeorg/test/endpoint"
 )
 
-var supported = map[string][]string{
-	"windows": {"powershell.exe", "-c", "Invoke-RestMethod -UseBasicParsing -Uri ('http://ipinfo.io/'+ (Invoke-WebRequest -UseBasicParsing -uri 'http://ifconfig.me/ip').Content)"},
-	"darwin":  {"bash", "-c", "wget -qO- http://ifconfig.me/ip | wget -qO- http://ipinfo.io/$1"},
-	"linux":   {"bash", "-c", "wget -qO- http://ifconfig.me/ip | wget -qO- http://ipinfo.io/$1"},
+func getPublicIP() (query string, lat float64, long float64, err error) {
+	response, err := http.Get("http://ip-api.com/json")
```
Comment (privateducky) [inline tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go:19]: terms/conditions or rate limiting concerns?

File: tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go [hunk 3]
```diff
@@ -6,20 +6,77 @@ CREATED: 2023-01-21
...
 }
 
 func test() {
-	command := supported[runtime.GOOS]
-	response := Endpoint.Shell(command)
-	print(response)
+	println("[+] Getting Public IP Address & Location")
+	query, lat, long, err := getPublicIP()
+	if err != nil {
+		println("Error while retrieving public IP")
```
Comment (privateducky) [inline tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go:69]: shouldn't the test be over at this point?

File: tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go [hunk 4]
```diff
@@ -6,20 +6,77 @@ CREATED: 2023-01-21
...
+	}
+
+	println("Public IP:", query)
+
+	fmt.Printf("Latitude: %.4f\n", lat)
```
Comment (privateducky) [inline tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go:74]: why are you using multiple formats to print to standard out? only use println please.

File: tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go (hunk 3 above)
Comment (privateducky) [reply tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go:69]: if yes; we don't need to return 'err' from the called function, instead that function can stop the test upon an error.

File: tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go [hunk 5]
```diff
@@ -6,20 +6,77 @@ CREATED: 2023-01-21
...
+	response, err := http.Get(url)
+	if err != nil {
+		println("Error while searching IP on Shodan:", err)
+		return
+	}
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		println("Error while reading response body:", err)
```
Comment (privateducky) [inline tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go:50]: same as IP test; I would think the test is over on an error here

File: tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go [hunk 6]
```diff
@@ -6,20 +6,77 @@ CREATED: 2023-01-21
...
+		return
+	}
+
+	if strings.Contains(string(body), "Ports open: ") {
```
Comment (privateducky) [inline tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go:54]: doing a string comparison is too brittle. figure out a way to structure the response data.

File: tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go [hunk 7]
```diff
@@ -6,20 +6,77 @@ CREATED: 2023-01-21
...
+	}
+
+	println("Public IP:", query)
+
+	fmt.Printf("Latitude: %.4f\n", lat)
+	fmt.Printf("Longitude: %.4f\n", long)
+
+	println("[+] Searching IP on Shodan")
+	searchShodan(query)
+
```
Comment (privateducky) [inline tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go:79]: this test is too open-ended in what "success" looks like.

there are several pieces of information collected:
1. ip
2. long/latt 
3. open ports (or even discoverability)

which of these criteria, or combination, mean the test should fail?

File: tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go [hunk 8]
```diff
@@ -6,20 +6,77 @@ CREATED: 2023-01-21
...
 import (
-	"github.com/preludeorg/test/endpoint"
-	"runtime"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	Endpoint "github.com/preludeorg/test/endpoint"
```
Comment (privateducky) [inline tests/dd270c6f-a41c-4115-b54d-ff940abd9c27/dd270c6f-a41c-4115-b54d-ff940abd9c27.go:15]: not shown here: but the test comment, specifically the "NAME" field in the comment, is out of date

**Review by privateducky — CHANGES_REQUESTED** [review]
several questions/concerns

**Comment by bfuzzy1** [comment]
Closing in favor of 2 separate VSTs. One that ping sweeps and another for arp scanning. It doesn't make sense to make changes to the original that moves it further away from the original idea of the test.

**Pull request closed without merging**